work_queue/src/work_queue_process.c

```diff
@@ -256,6 +260,60 @@ pid_t work_queue_process_execute(struct work_queue_process *p )
 		if(result == -1)
 			fatal("could not dup pipe to stderr: %s", strerror(errno));
 
+		if(strcmp(p->task->command_line, "@FUNCTION") == 0) {
+			// this code simply prints some basic json to the process output file
+			/*
+			char * result_json = "{\"result\": \"success\"}";
+			write(p->output_fd, result_json, strlen(result_json));
+			close(p->output_fd);
+			exit(0);
+			*/
+			
+			// Should this be split into a separate function?
+			// i.e. void work_queue_process_network_function(struct work_queue_process *p)
+			// build necessary networking components
+			struct hostent *hp;
```

Comment:
This is the right idea, but please use the modules in `dttools` for dealing with the network,
which already deal with a variety of cross platform issues, ipv6, and so forth:

`link` provides an abstractions for network connections, and handles 
https://github.com/cooperative-computing-lab/cctools/blob/master/dttools/src/link.h

`domain_name` performs domain name lookups:
https://github.com/cooperative-computing-lab/cctools/blob/master/dttools/src/domain_name.h

Reply:
I have started to change the code to use `link` but I am having some trouble with `link_connect` throwing a connection refused error. I'm running `network_function.py` on the same machine as the worker and the port is hardcoded right now, so I pushed up the changes to see if you see something that looks wrong.

Reply:
Aha, your server is listening on a udp port, but the link module does tcp.
Use the `datagram` module instead if you want to do udp instead.

Reply:
Fixed!